x-pack/plugin/sql/qa/single-node/src/test/java/org/elasticsearch/xpack/sql/qa/single_node/JdbcCsvSpecIT.java

```diff
@@ -8,8 +8,25 @@
 import org.elasticsearch.xpack.sql.qa.jdbc.CsvSpecTestCase;
 import org.elasticsearch.xpack.sql.qa.jdbc.CsvTestUtils.CsvTestCase;
 
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
 public class JdbcCsvSpecIT extends CsvSpecTestCase {
     public JdbcCsvSpecIT(String fileName, String groupName, String testName, Integer lineNumber, CsvTestCase testCase) {
         super(fileName, groupName, testName, lineNumber, testCase);
     }
+
+    @Override
+    protected ResultSet executeJdbcQuery(Connection con, String query) throws SQLException {
+        // using a smaller fetchSize for nested documents' tests to uncover bugs
+        // similar with https://github.com/elastic/elasticsearch/issues/35176 quicker
+        if (fileName.startsWith("nested")) {
```

Comment:
Instead of overriding `executeJdbcQuery`, override `fetchSize()`:

```java
protected int fetchSize() {
        if (fileName.startsWith("nested")) {
           return randomBoolean() ? super.fetchSize() : randomIntBetween(1,5));
        }
  }
```